### PR TITLE
Add cargo-make file with static build commands + small fixes

### PR DESCRIPTION
**Pull request by ZetaTwo**
Thanks for building this tool! I have a scenario where I need to be able to quickly deploy this on a heterogeneous set of hosts so I looked into the possibility of building shh as a static binary for easier deployment. It turns out that this works great with the right build flags.

I added a [cargo-make](https://sagiegurari.github.io/cargo-make/) file with the commands so that it's easy to build a static binary by running `cargo make build-static-release`. I also made sure to run the tests which can also be run with `cargo make test-all-static` and finally I went through the Caddy example from the article by first building the binary on my Ubuntu machine, uploading it to a Debian 12 machine and going through the example which worked great.

I also took the opportunity to update the dependencies ~~and apply a small clippy suggestion~~.

### CODE REVIEW

**Comment by desbma** [comment]
Thanks for the contribution!

This PR contains several different things, that would be better handled in different PRs.

I'll comment on each change anyway.

### Updated dependencies

I try to avoid blindly updating dependencies without a specific reason. Is there a specific reason you bumped the versions?
When I tag a new release, I usually run `cargo audit` to check for known vulnerabilities or abandoned crates, and `cargo update` to update minor versions but I keep the same semver requirements in `Cargo.toml` to avoid unexpected breakage, and run a bunch of manual tests on different systems. Unless there is a good reason (like new features I want), I avoid bumping minor or major versions in between to limit noise.

## Cargo make

I don't know much about `cargo make`, but try to avoid directly depending on third party tools, unless they are absolutely necessary. For example this makes distro packaging more difficult.

Regarding the need to have "portable" static builds: I have the same use case. For example the Debian packages are build with musl, see https://github.com/desbma/shh/blob/master/.github/workflows/release.yml. In my experience, passing `--target x86_64-unknown-linux-musl` to cargo commands, after installing the target with rustup is enough to make 100% static (don't even link dynamically to libc) binaries.

## Readme addition

Looks good to me.
Out of curiosity, what strace version did you use?

## Clippy fix

Thanks, I have automatically applied this fix and some others in 734c0901f9156a1d995a6d4b60370e2aeeba6ece.

**Comment by ZetaTwo** [comment]
> This PR contains several different things, that would be better handled in different PRs.

Yes. Sorry about that.

> Unless there is a good reason (like new features I want), I avoid bumping minor or major versions in between to limit noise.

Very fair. I like to keep things generally updated to have a smaller difference when I actually have to update but I completely understand the other side as well.

> third party tools, unless they are absolutely necessary.

Yeah, I just wanted to store the build commands used for the static builds so that I don't have to look up the flags every time. I could do this in a regular makefile, in the readme or not at all.

>  In my experience, passing --target x86_64-unknown-linux-musl to cargo commands

Yes, this works but I wanted to build statically with glibc and then you need that env variable as well. The resulting binary is fully static.

> Out of curiosity, what strace version did you use?

I think I first used 6.1.? which was what Debian 12 had. Then I hid that bug and built 6.8 from source before I realized it was a parsing bug.

> Thanks, I have automatically applied this fix and some others

Nice!

So, in summary:
- Can/should I put the static build commands in README or a regular Makefile
- Other than that, the readme update is the only interesting.

**Comment by desbma** [comment]
> Very fair. I like to keep things generally updated to have a smaller difference when I actually have to update but I completely understand the other side as well.

What I will probably do in the future instead of relying on my ~crappy~minimalistic release script, is to try something like [https://github.com/MarcoIeni/release-plz](release-plz).

> Yes, this works but I wanted to build statically with glibc and then you need that env variable as well. The resulting binary is fully static.

I think we want/need the same things: fully static portable Linux binaries. 
Currently I automatically build and publish Debian packages with the CI pipeline. Maybe you need support for other distributions than Debian? 
Is there a concrete benefit from using glibc instead of musl? I have heard it results in huge binaries but I have not tested it myself.
If things can be improved regarding the static builds, I'd rather work on the current CI pipeline so that the binaries/packages people need are automatically built, rather than add instructions to the README that are not specific to this project. 
This way users who want to quickly download binaries can do it, and those who want to build themselves or customize the code, either already know how to do it, or can look at the CI scripts.

**Comment by ZetaTwo** [comment]
I don't think I had a very concrete specific reason to prefer glibc over musl other than just some vague feeling that it's more known and tested library. I also avoided installing another toolchain for it but that's also a very minor argument.

I could add another task to the CI so that it builds both a glibc and a musl version if that's preferable.

**Comment by desbma** [comment]
Apparently the target `x86_64-unknown-linux-gnu` is a tier 1 platform, and `x86_64-unknown-linux-musl` is tier 2 so explicitly less tested.
Cf. https://doc.rust-lang.org/stable/rustc/platform-support.html

This is good enough reason to change, I have changed the CI script to use glibc, see 09e6f66d8d35139350b0b442b1e3696a41560381.

**Comment by desbma** [comment]
You can keep the commit that updates the README and remove the rest, I will then merge it and make a new release.

Thanks again

**Comment by ZetaTwo** [comment]
Nice. Done. Happy to help with the project.
